### DISCUSSION TITLE
Artillery retrying requests on 403 response

### DISCUSSION THREAD

**Discussion by carrejoe3** [artilleryio/artillery#2436] [Artillery]
Hey, we're running some load tests using artillery and noticing way more requests being sent when we implement rate limiting and start returning 403 responses whenever the max RPS is met. Is this because artillery automatically retries requests?

**Comment by hassy** [comment]
hi @carrejoe3 :wave: if you're looking at the increase in `http.request_rate` metric, it's most likely because 403 responses have a low response time. The number of requests made by Artillery is a function of two things: (1) load phase config of the Artillery test, and (2) how responsive the system under test is. So a system that responds faster will result in more requests sent per second. Hope that make sense

**Reply by hassy** [reply]
oh and just to confirm - Artillery does not retry requests

**Reply by carrejoe3** [reply]
Hey @hassy yes that makes sense, thanks for clarifying. The thing is though, why not just use the RPS which is specified in the config, regardless of how quick the responses are?

**Reply by hassy** [reply]
Ah that's a common point of misunderstanding (we definitely need to figure out a way to make it clearer in the docs). Artillery's load config is centered around virtual users, not requests. As such Artillery does not allow you to configure constant RPS. It doesn't really make sense for the type of testing Artillery is designed for. So in Artillery's load model virtual users respond to the backpressure from the server - much like they would in the real world.

More background in this blog post: https://www.artillery.io/blog/load-testing-workload-models

**Reply by carrejoe3** [reply]
great okay thanks for the info @hassy
